shared/fs/filepreview/index.js

```diff
@@ -0,0 +1,117 @@
+// @flow
+import * as React from 'react'
+import * as Types from '../../constants/types/fs'
+import {globalStyles, globalColors, globalMargins, isMobile} from '../../styles'
+import {Box, Button, Icon, Text} from '../../common-adapters'
+import {formatTimeForMessages} from '../../util/timestamp'
+import FolderHeader from '../header/header-container'
+
+type FilePreviewHeaderProps = {
+  title: string,
+  desc: string,
+}
+
+const FilePreviewHeader = ({title, desc}: FilePreviewHeaderProps) => (
+  <Box>
+    <Box style={filePreviewHeaderStyle}>
+      <Text type="BodyBig">{title}</Text>
+      <Text type="BodySmall">{desc}</Text>
+    </Box>
+  </Box>
+)
+
+type FilePreviewProps = {
+  path: Types.Path,
+  meta: Types.PathItemMetadata,
+}
+
+const humanReadableFileSize = meta => {
+  const kib = 1024
+  const mib = kib * kib
+  const gib = mib * kib
+  const tib = gib * kib
+
+  if (!meta) return ''
+  const size = meta.size
+  if (size >= tib) return `${Math.round(size / tib)}tb`
+  if (size >= gib) return `${Math.round(size / gib)}gb`
+  if (size >= mib) return `${Math.round(size / mib)}mb`
+  if (size >= kib) return `${Math.round(size / kib)}kb`
+  return '' + size
+}
+
+class FilePreview extends React.PureComponent<FilePreviewProps> {
+  render() {
+    const {path, meta} = this.props
+    const fileName = Types.getPathName(path)
+    let desc = 'loading'
+    if (meta) {
+      desc = 'Modified on ' + formatTimeForMessages(meta.lastModifiedTimestamp)
+      if (meta.lastWriter) desc += ' by ' + meta.lastWriter
+    }
+    return (
+      <Box style={styleOuterContainer}>
+        <Box style={globalStyles.flexBoxColumn}>
+          <FolderHeader path={path} />
+        </Box>
+        <FilePreviewHeader title={fileName} desc={desc} />
+        <Box style={stylesContainer}>
+          <Icon type={iconTypeName} />
```

Comment:
It seems we could use [`PathItemIcon`](https://github.com/keybase/client/blob/ab2ec42137552a9c6666447949c64ee23d53bba7/shared/fs/row/path-item-icon.js) here. It's in `row` now but if I can move it to a different dir later if we decide to use it here.

Reply:
Noted in a comment for now